custom_components/mypyllant/climate.py

```diff
@@ -642,3 +666,174 @@ async def async_set_preset_mode(self, preset_mode):
+            "button_lock": self.room.room_configuration.button_lock,
+        }
+        return attr | self.room.extra_fields
+
+    async def set_quick_veto(self, **kwargs):
+        _LOGGER.debug("Setting quick veto on %s with params %s", self.room.name, kwargs)
+        temperature = kwargs.get("temperature")
+        duration_hours = kwargs.get("duration_hours")
+        await self.coordinator.api.quick_veto_ambisense_room(
+            self.room, temperature, duration_hours, self.default_quick_veto_duration
+        )
+        await self.coordinator.async_request_refresh_delayed()
+
+    async def remove_quick_veto(self):
+        _LOGGER.debug("Removing quick veto on %s", self.room.name)
+        await self.coordinator.api.cancel_quick_veto_ambisense_room(self.room)
+        await self.coordinator.async_request_refresh_delayed()
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Return the list of supported features."""
+        return (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.TURN_OFF
+            | ClimateEntityFeature.TURN_ON
+        )
+
+    @property
+    def target_temperature(self) -> float | None:
+        return self.room.room_configuration.temperature_setpoint
+
+    @property
+    def current_temperature(self) -> float | None:
+        return self.room.room_configuration.current_temperature
+
+    @property
+    def current_humidity(self) -> float | None:
+        return self.room.room_configuration.current_humidity
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        return AMBISENSE_ROOM_OPERATION_MODE_MAP.get(
+            self.room.room_configuration.operation_mode, HVACMode.OFF
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode):
+        mode = [
+            k for k, v in AMBISENSE_ROOM_OPERATION_MODE_MAP.items() if v == hvac_mode
+        ][0]
+        await self.coordinator.api.set_ambisense_room_operation_mode(self.room, mode)
+        await self.coordinator.async_request_refresh_delayed()
+
+    @property
+    def preset_mode(self) -> str | None:
+        if self.room.room_configuration.quick_veto_end_time is not None:
+            return PRESET_BOOST
+        else:
+            return PRESET_NONE
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        if preset_mode == PRESET_NONE:
+            await self.remove_quick_veto()
+        # Both preset none and boost exist, but going from none to boost makes no sense without a specific
+        # target temperature
+        self._valid_mode_or_raise("preset", preset_mode, [PRESET_NONE])
+
+    async def async_turn_on(self) -> None:
+        await self.async_set_hvac_mode(self.data["last_active_hvac_mode"])
+
+    async def async_turn_off(self) -> None:
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
```

Comment:
Do you want the same implementation as for zone when chaging target temp ?

I don't think we should update the time program for a room. First we need to find the correct `RoomTimeProgramDay` and set `temperature_setpoint` accordingly to the new value. But then, should we also update `RoomTimeProgramDay` for other days ?  I think it will be misleading for users. Moreover, in the mobile app, when you just change the temperature, it is creating a quick veto.

What I propose:
- If manual, simply update the target temp (I will need to find the correct endpoint for that) -> Same behavior as zone and mobile app
- Otherwise, create quick veto -> same behavior as mobile app (and as a zone, if the option is checked)

Reply:
> I don't think we should update the time program for a room.

Makes sense. I added the schedule overwrite as an option that's off by default on zones, so what happens in the app and what you describe matches the default behavior on zones. Nowadays there's also a calendar entity for schedule changes, so I'm happy to skip the overwrite option. It's not the cleanest logic, like you said.

A better fit for the schedule is probably another calendar entity. Those already exist for zone & water heater schedules, and should be easy enough to add for rooms. What tripped me up, is that there seems to be no `endTime` on room time programs in your test data - do you know what that is about?

Reply:
Indeed, there is no end time, but *actual* end time is the start time of the next `RoomTimeProgramDay`. And of course the whole day needs to be "covered" by a time window, since there is no fallback temperature for a room in AUTO mode.

I don't know why they did such different mechanism for rooms and zones, but honestly I find the rooms way much more user friendly and actually much easier to handle from technical point of view. 

This is the logic in pymultimatic: https://github.com/thomasgermain/pymultiMATIC/blob/master/pymultimatic/model/timeprogram.py#L137 (funny thing is that I had to find the current time program to find the actual target temperature for a room, there was no aggregate field for that in the API). 

The logic looks quite complex, but the idea is to iterate over all 'time windows" and find the one which is after the `search_date`.

Reply:
And this is how it looks like in the mobile app
![Screenshot_2024-03-19-23-42-20-00_ab6c4a494f5f354113de683b4ea3527b](https://github.com/signalkraft/mypyllant-component/assets/12560542/723a251a-fdbd-485f-96a0-e350b51ac2ad)